drivers/serial/uart_mcux_lpuart.c

```diff
@@ -616,37 +616,22 @@ static int uart_mcux_lpuart_dma_replace_rx_buffer(const struct device *dev)
 	struct mcux_lpuart_data *data = (struct mcux_lpuart_data *)dev->data;
 	const struct mcux_lpuart_config *config = dev->config;
 	LPUART_Type *lpuart = config->base;
-	struct mcux_lpuart_rx_dma_params *rx_dma_params = &data->async.rx_dma_params;
+	// struct mcux_lpuart_rx_dma_params *rx_dma_params = &data->async.rx_dma_params;
 
 	LOG_DBG("Replacing RX buffer, new length: %d", data->async.next_rx_buffer_len);
 
 	/* There must be a buffer to replace this one with */
 	assert(data->async.next_rx_buffer != NULL);
 	assert(data->async.next_rx_buffer_len != 0U);
-	rx_dma_params->buf = data->async.next_rx_buffer;
-	rx_dma_params->buf_len = data->async.next_rx_buffer_len;
-	rx_dma_params->offset = 0;
-	rx_dma_params->counter = 0;
-	data->async.next_rx_buffer = NULL;
-	data->async.next_rx_buffer_len = 0U;
 
 	const int success =
 		dma_reload(config->rx_dma_config.dma_dev, config->rx_dma_config.dma_channel,
-			   LPUART_GetDataRegisterAddress(lpuart), (uint32_t)rx_dma_params->buf,
-			   rx_dma_params->buf_len);
+			   LPUART_GetDataRegisterAddress(lpuart), (uint32_t)data->async.next_rx_buffer,
```

Comment:
you change the logic: here:
old:
dma_reload next buffer, request buffer update.
now:
request buffer update, so the next is replaced before use, and then you dma_reload the net buffer, which means the original next buffer is not used at all.

Reply:
yes, I reverted part of the code in your last PR. it is because we have to remain buf/buf_len until DMA callback, at the same time, we need submit next_rx_buf to DMA.  The original logic (before your PR  https://github.com/zephyrproject-rtos/zephyr/pull/79325/files) works well. So I use it as before.